core-server/src/main/java/org/glassfish/jersey/server/spi/ServerProvider.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) 2018 Markus KARG. All rights reserved.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0, which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception, which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ */
+
+package org.glassfish.jersey.server.spi;
+
+import javax.ws.rs.ConstrainedTo;
+import javax.ws.rs.JAXRS;
+import javax.ws.rs.ProcessingException;
+import javax.ws.rs.RuntimeType;
+import javax.ws.rs.core.Application;
+
+import org.glassfish.jersey.server.ApplicationHandler;
+import org.glassfish.jersey.spi.Contract;
+
+/**
+ * Service-provider interface for creating server instances.
+ *
+ * If supported by the provider, a server instance of the requested Java type
+ * will be created.
+ * <p>
+ * The created server uses an internally created {@link Container} which is
+ * responsible for listening on a communication channel provided by the server
+ * for new client requests, dispatching these requests to the registered
+ * {@link ApplicationHandler Jersey application handler} using the handler's
+ * {@link ApplicationHandler#handle(org.glassfish.jersey.server.ContainerRequest)
+ * handle(requestContext)} method and sending the responses provided by the
+ * application back to the client.
+ * </p>
+ * <p>
+ * A provider shall support a one-to-one mapping between a type, provided the
+ * type is not {@link Object}. A provider may also support mapping of sub-types
+ * of a type (provided the type is not {@code Object}). It is expected that each
+ * provider supports mapping for distinct set of types and subtypes so that
+ * different providers do not conflict with each other. In addition, a provider
+ * SHOULD support the super type {@link Server} to participate in auto-selection
+ * of providers (in this case the <em>first</em> supporting provider found is
+ * used).
+ * </p>
+ * <p>
+ * An implementation can identify itself by placing a Java service provider
+ * configuration file (if not already present) -
+ * {@code org.glassfish.jersey.server.spi.ServerProvider} - in the resource
+ * directory {@code META-INF/services}, and adding the fully qualified
+ * service-provider-class of the implementation in the file.
+ * </p>
+ *
+ * @author Markus KARG (markus@headcrashing.eu)
+ * @since 2.28
```

Comment:
2.28 is EE4J_8

Reply:
What is the planned Jersey version number for adopting JAX-RS 2.2?

Reply:
"ping"? ;-)

Reply:
It actually depends on when JAX-RS 2.2 is to be released. 2.28 should be released this year. 2.29 contains a bunch of fixes that we would like to release, too. Should JAX-RS 2.2 be released around 2.29 time, 2.30 sounds just right

Reply:
Understood. Sad but true, the release date of JAX-RS 2.2 in turn is mostly dependend on the missing contract between Oracle and Eclipse Foundation. So all we can do is to simply keep this PR open, and as soon as the contract is there I will strive for releasing JAX-RS 2.2 and then update the `@since` markers.

Reply:
As JAX-RS 2.1.4 was published two weeks ago, but 2.28 is not published still, I'd like to ask again, how to go on with this topic? I would like to speed up the pace of JAX-RS's new features now that 2.1.4 seems to be the final release.

Reply:
We have a [plan](https://github.com/eclipse-ee4j/jersey/wiki/Road-Map) to support this bootstrap in 2.30, sometimes in September. Does it sound good for you?

Reply:
The version "Jersey 2.30" and "having a public plan" sounds really great, but waiting until September makes me cry. ;-)

Reply:
I pushed another commit replacing 2.29 by 2.30, so this discussion is resolved. :-)